bakerydemo/base/wagtail_hooks.py

```diff
@@ -74,3 +83,72 @@ class BakeryModelAdminGroup(ModelAdminGroup):
 # you only need to register the ModelAdminGroup class with Wagtail:
 modeladmin_register(BreadModelAdminGroup)
 modeladmin_register(BakeryModelAdminGroup)
+
+if settings.PREVENT_ADMIN_CREDENTIALS_CHANGE:
+
+    @hooks.register("after_edit_user")
+    def notify_rejected_admin_changes(request, user):
+        """
+        Notify the user when they attempt to change the admin user's password that it won't work.
+
+        See the `prevent_user_password_change` signal below for where the noop is done.
+        """
+
+        if user.id != settings.DEFAULT_ADMIN_PK:
+            return
+
+        if request.POST.get("password1"):
+            messages.add_message(
+                request,
+                messages.ERROR,
+                "You can't change the admin user's password. The password change was ignored.",
+            )
+
+        if (
+            request.POST.get("username", settings.DEFAULT_ADMIN_USERNAME)
+            != settings.DEFAULT_ADMIN_USERNAME
+        ):
+            messages.add_message(
+                request,
+                messages.ERROR,
+                "You can't change the admin user's username. The username change was ignored.",
+            )
+
+    @receiver(pre_save, sender=User)
+    def prevent_admin_changes(sender, instance, *args, **kwargs):
+        """
+        Prevent the "admin" user's password from being changed.
+        """
+
+        if instance.id != settings.DEFAULT_ADMIN_PK:
+            return
+
+        # Check that `set_password` was called
+        if instance._password:
+            instance.set_password(settings.DEFAULT_ADMIN_PASSWORD)
+
+        if instance.username != settings.DEFAULT_ADMIN_USERNAME:
+            instance.username = settings.DEFAULT_ADMIN_USERNAME
```

Comment:
We should also block attempts to unset the `is_superuser` and `is_active` flags here.